compiler/x/codegen/ControlFlowEvaluator.cpp

```diff
@@ -1298,11 +1298,16 @@ TR::Register *OMR::X86::TreeEvaluator::iternaryEvaluator(TR::Node *node, TR::Cod
    TR::Register *falseReg = cg->evaluate(falseVal);
    bool trueValIs64Bit = TR::TreeEvaluator::getNodeIs64Bit(trueVal, cg);
    TR::Register *trueReg  = TR::TreeEvaluator::intOrLongClobberEvaluate(trueVal, trueValIs64Bit, cg);
+   if (!node->isNotCollected()) 
+      trueReg->setContainsCollectedReference();
 
    // don't need to test if we're already using a compare eq or compare ne
    auto conditionOp = condition->getOpCode();
+   bool longCompareOn32bit = (conditionOp.isBooleanCompare() &&
+         (condition->getFirstChild()->getOpCode().isLong() || condition->getSecondChild()->getOpCode().isLong())
+         && (!TR::Compiler->target.is64Bit()));
    //if ((conditionOp == TR::icmpeq) || (conditionOp == TR::icmpne) || (conditionOp == TR::lcmpeq) || (conditionOp == TR::lcmpne))
-   if (conditionOp.isCompareForEquality())
+   if (conditionOp.isCompareForEquality() && condition->getFirstChild()->getOpCode().isIntegerOrAddress() && !longCompareOn32bit)
```

Comment:
Similarly, it may be worthy bring `!longCompareOn32bit` as the first condition, so that the rest checks can be skipped when `!longCompareOn32bit` is false.

Reply:
addressed.